libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -753,6 +759,29 @@ bool AP_BattMonitor::healthy() const
     return true;
 }
 
+void AP_BattMonitor::handle_mavlink_battery_message(const mavlink_message_t &msg) {
+    switch (msg.msgid) {
+        case  MAVLINK_MSG_ID_BATTERY_STATUS: 
+            select_battery_driver(msg, mavlink_msg_battery_status_get_id(&msg));   
```

Comment:
The battery ID is not the driver I'd. It should match the battery serial number param or if the serial number is negative. This check is done in the driver. Just send this packet to all backends like all the other MAV drivers do. The empty virtual function calls won't do anything.